jurity/recommenders/interlist_diversity.py

```diff
@@ -0,0 +1,214 @@
+# -*- coding: utf-8 -*-
+# Copyright FMR LLC <opensource@fidelity.com>
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import List, Union, Tuple
+
+import numpy as np
+import pandas as pd
+import scipy.sparse as sp
+from sklearn.metrics.pairwise import cosine_similarity
+import warnings
+
+from jurity.utils import Constants
+
+
+def sample_users(df: pd.DataFrame, user_id_column: str = Constants.user_id,
+                 sample_size: float = None, seed: int = Constants.default_seed) -> pd.DataFrame:
+    """
+    Samples input data frame by selecting a random sample of users.
+
+    Parameters
+    ----------
+    df: pd.DataFrame
+        Data frame with a user_id_col column.
+    user_id_column: str
+        User id column name.
+    sample_size: float
+        Proportion of users to randomly sample for evaluation.
+    seed : int, default=Constants.default_seed
+        The seed used to create random state.
+
+    Returns
+    -------
+    Sampled data frame
+    """
+    rng = np.random.RandomState(seed)
```

Comment:
Could you use `default_rng` here, since it's now the default in numpy? (and I'm hoping our numpy dependency is high enough to support it :D)

Reply:
@dorukkilitcioglu  what would be the difference? To make sure we are using something newer?

Reply:
Yeah, basically not introducing any tech debt while we have the chance to address it.

Reply:
So according to the numpy docs, this `np.random.default_rng()` would serve as the replacement for `RandomState`. Additional generator component and future algorithmic improvements would be reserved for the new class.